src/datasource/components/QueryEditor/HostTagQueryEditor.tsx

```diff
@@ -0,0 +1,137 @@
+    setHostTagFilters((prevFilters) => [
+      ...prevFilters,
+      { tag: '', value: '', operator: HostTagOperatorValue.Contains },
+    ]);
+  }, []);
+
+  const onRemoveHostTagFilter = useCallback((index: number) => {
+    setHostTagFilters((prevFilters) => prevFilters.filter((_, i) => i !== index));
+  }, []);
+
+  const setHostTagFilterName = useCallback((index: number, name: string) => {
+    setHostTagFilters((prevFilters) =>
+      prevFilters.map((filter, i) => (i === index ? { ...filter, tag: name } : filter))
+    );
+  }, []);
+
+  const setHostTagFilterValue = useCallback((index: number, value: string) => {
+    if (value !== undefined) {
+      setHostTagFilters((prevFilters) =>
+        prevFilters.map((filter, i) => (i === index ? { ...filter, value: value } : filter))
+      );
+    }
+  }, []);
+
+  const setHostTagFilterOperator = useCallback((index: number, operator: HostTagOperatorValue) => {
+    setHostTagFilters((prevFilters) =>
+      prevFilters.map((filter, i) => (i === index ? { ...filter, operator } : filter))
+    );
+  }, []);
+
+  useEffect(() => {
+    onHostTagFilterChange(hostTagFilters);
+  }, [hostTagFilters]);
+  return (
+    <div>
+      <Stack direction="row">
+        <Tooltip content="Add host tag filter">
+          <Button icon="plus" variant="secondary" aria-label="Add new host tag filter" onClick={onAddHostTagFilter} />
+        </Tooltip>
+        {hostTagFilters.length > 0 && (
+          <RadioButtonGroup
+            options={[
+              { label: 'AND/OR', value: '0' }, // Default
+              { label: 'OR', value: '2' },
+            ]}
+            onChange={onHostTagEvalTypeChange}
+            value={evalTypeValue ?? '0'}
+          />
+        )}
+      </Stack>
+      <Stack direction="column">
+        {hostTagFilters.map((filter, index) => {
+          return (
+            <Stack key={`host-tag-filter-${index}`} direction="row">
+              <Combobox
+                value={filter.tag}
+                onChange={(option: ComboboxOption) => setHostTagFilterName(index, option.value)}
+                options={hostTagOptions ?? []}
+                width={19}
+                loading={hostTagOptionsLoading}
+              />
+              <Combobox
+                value={filter.operator}
+                onChange={(option: ComboboxOption<HostTagOperatorValue>) =>
+                  setHostTagFilterOperator(index, option.value)
+                }
+                options={operatorOptions}
+                width={19}
+              />
+              {filter.operator !== HostTagOperatorValue.Exists &&
+                filter.operator !== HostTagOperatorValue.DoesNotExist && (
+                  <Input
+                    value={filter.value}
+                    onChange={(evt: FormEvent<HTMLInputElement>) =>
+                      setHostTagFilterValue(index, evt?.currentTarget?.value)
+                    }
+                    width={19}
+                  />
```

Comment:
A short placeholder would be useful. At first glance I've missed what I need to do with the input field.